tests/Integration/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/WebserviceControllerTest.php

```diff
@@ -0,0 +1,167 @@
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\PrestaShopBundle\Controller\Admin\Configure\AdvancedParameters;
+
+use Symfony\Bundle\FrameworkBundle\KernelBrowser;
+use Symfony\Bundle\FrameworkBundle\Routing\Router;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+use Symfony\Component\HttpFoundation\Session\Session;
+use WebserviceKey;
+
+class WebserviceControllerTest extends WebTestCase
+{
+    /**
+     * @var KernelBrowser
+     */
+    protected $client;
+    /**
+     * @var Router
+     */
+    protected $router;
+    /**
+     * @var Session
+     */
+    protected $session;
+    /**
+     * @var WebserviceKey
+     */
+    protected $webserviceKey;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        self::bootKernel();
+
+        $this->client = self::createClient();
+        $this->router = self::$kernel->getContainer()->get('router');
+        $this->session = self::$kernel->getContainer()->get('session');
+
+        $this->webserviceKey = new WebserviceKey();
+        $this->webserviceKey->key = 'DFS51LTKBBMBAF5QQRG523JMQYEHA4X7';
+        $this->webserviceKey->description = 'Description WS Key';
+        $this->webserviceKey->active = true;
+        $this->webserviceKey->add();
+    }
+
+    public function tearDown(): void
+    {
+        parent::tearDown();
+
+        $this->webserviceKey->delete();
+    }
+
+    /**
+     * @dataProvider dataProviderBulkEnable
+     */
+    public function testBulkEnable(bool $actual): void
+    {
+        $this->webserviceKey->active = $actual;
+        $this->webserviceKey->save();
+
+        $this->client->request(
+            'POST',
+            $this->router->generate(
+                'admin_webservice_keys_bulk_enable'
+            ),
+            [
+                'webservice_key_bulk_action' => [
+                    $this->webserviceKey->id,
+                ],
+            ]
+        );
+        $response = $this->client->getResponse();
+        $this->assertTrue($response->isRedirection());
+
+        // Check session
+        $all = $this->session->getFlashBag()->all();
+        $this->assertArrayHasKey('success', $all);
+        $this->assertSame('The status has been successfully updated.', $all['success'][0]);
+
+        // Check status
+        $webserviceKey = new WebserviceKey($this->webserviceKey->id);
+        $this->assertTrue((bool) $webserviceKey->active);
+    }
+
+    public function dataProviderBulkEnable(): array
```

Comment:
Since you have only a boolean behaviour, you can use the same data provider.

Reply:
Fixed